great_expectations/datasource/fluent/pandas_datasource.py

```diff
@@ -503,49 +507,85 @@ class PandasDatasource(_PandasDatasource):
     def test_connection(self, test_assets: bool = True) -> None:
         ...
 
+    @staticmethod
+    def _validate_asset_name(asset_name: Optional[str] = None) -> str:
+        if asset_name == DEFAULT_PANDAS_DATA_ASSET_NAME:
+            raise PandasDatasourceError(
+                f"""An asset_name of {DEFAULT_PANDAS_DATA_ASSET_NAME} cannot be passed because it is a reserved name."""
+            )
+        if not asset_name:
+            asset_name = DEFAULT_PANDAS_DATA_ASSET_NAME
+        return asset_name
+
     def _get_validator(self, asset: _PandasDataAsset) -> Validator:
         batch_request: BatchRequest = asset.build_batch_request()
         return self._data_context.get_validator(batch_request=batch_request)  # type: ignore[arg-type] # got BatchRequest expected BatchRequestBase
 
-    def add_dataframe_asset(self, name: str, dataframe: pd.DataFrame) -> DataFrameAsset:
+    def add_dataframe_asset(
+        self,
+        name: str,
+        dataframe: pd.DataFrame,
+        batch_metadata: Optional[BatchMetadata] = None,
+    ) -> DataFrameAsset:
         asset = DataFrameAsset(
             name=name,
             dataframe=dataframe,
+            batch_metadata=batch_metadata,
         )
         return self._add_asset(asset=asset)
 
     def read_dataframe(
-        self, dataframe: pd.DataFrame, asset_name: Optional[str] = None
+        self,
+        dataframe: pd.DataFrame,
+        asset_name: Optional[str] = None,
+        batch_metadata: Optional[BatchMetadata] = None,
     ) -> Validator:
-        if not asset_name:
-            asset_name = DEFAULT_PANDAS_DATA_ASSET_NAME
+        name: str = self._validate_asset_name(asset_name=asset_name)
         asset: DataFrameAsset = self.add_dataframe_asset(
-            name=asset_name, dataframe=dataframe
+            name=name,
+            dataframe=dataframe,
+            batch_metadata=batch_metadata,
         )
         return self._get_validator(asset=asset)
 
-    def add_clipboard_asset(self, name: str, **kwargs) -> ClipboardAsset:  # type: ignore[valid-type]
+    def add_clipboard_asset(
+        self,
+        name: str,
+        batch_metadata: Optional[BatchMetadata] = None,
+        **kwargs,
```

Comment:
```suggestion
        **kwargs,
```
I think `batch_metadata` could be part of the `**kwargs` and then it would never be `None`.

Reply:
I will make this change in a follow-up PR per our offline conversation.